test/e2e_flags.go

```diff
@@ -73,9 +82,48 @@ func initializeFlags() *EnvironmentFlags {
 
 	flag.StringVar(&f.Tag, "tag", "latest", "Provide the version tag for the test images.")
 
+	klog.InitFlags(nil)
+	flag.Set("v", "2") // A normal level of logging for testing
+
 	return &f
 }
 
+func printFlags() {
+	fmt.Print("Test Flags: {")
+	flag.CommandLine.VisitAll(func(f *flag.Flag) {
+		fmt.Printf("'%s': '%s', ", f.Name, f.Value.String())
```

Comment:
%q is equivalent to '%s', and slightly better than it in some cases
not sure what `"%+v", f` would do

Reply:
%q is "escaped with Go syntax" and I don't wish to alter the values as they aren't limited by Go syntax.

*Flag has several irrelevant children: usage, DefValue, and Value has two functions at children instead of actually printing the value: https://godoc.corp.google.com/pkg/flag#Flag

Reply:
I think `%q` is completely appropriate here.

Reply:
Please don't resolve until we settle on a decision.
I still don't understand the `%q` problem. Please bring an example that you think will make this worse? (

Reply:
Because '%s' is obvious to anybody reading the code, has ever used printf strings, and will do exactly what I want it to do. It's not easy to find what %q is going to do besides "escaped with Go syntax" -- whatever that means -- and uses double-quotes instead.

Reply:
Double quotes is not really a problem.
`%q` is part of the basic `fmt` package which is usually covered in the go walkthrough early pages.

Reply:
I have never encountered it before this comment and we're gaining nothing with %q that '%s' doesn't do just fine.

Reply:
It is at least 2 characters longer... :)

Reply:
coryrc@coryrc:~/go/src/knative.dev/pkg$ grep -r "'%s'" | wc -l
74

Reply:
A more proper command is `grep -r "'%s'" * | grep -v "vendor" | wc -l`, but it still returns 33 instances, which I guess I will need to fix :-)

Reply:
I hope you don't! :-)